### ISSUE TITLE
Since 1.11.0, can't ignore-eslint in JSX blocks (comment is ignored)

### ISSUE THREAD

**Issue by adrienharnay** [typescript-eslint/typescript-eslint#643]
**What code were you trying to parse?**

```ts
<SomeComponent
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  onSubmit={(value: any) => {}}
/>
```

**What did you expect to happen?**

Eslint should not produce an error for `value: any`

**What actually happened?**

It did :(. It works well with 1.10.2

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.11.0` |
| `TypeScript`                           | `3.5.2` |
| `node`                                 | `10.15.3` |
| `npm`                                  | `6.4.1` |

Thanks in advance!

**Comment by Gpx** [comment]
I'm having the same issues but outside JSX blocks

**Comment by bradzacher** [comment]
related #648

regressions introduced by recent changes to jsx comment handing

**Comment by nathanmarks** [comment]
I believe this is a bug in the typescript scanner.

Given this code:
```ts
const pure = () => {
  return (
      <Foo
        // one
        foo={123}
        // two
        bar="woof"
      >
      </Foo>
  );
}
```

the token it recognizes as a single line comment looks like this:

```
'// one'
```

But the second comment, which is not being recognised here, has a token that looks like this:

```
'}\n        // two\n        bar="woof"\n      >\n      '
```

and so the token is not being recognised properly.

**Comment by nathanmarks** [comment]
This is the suspect code causing the issue: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/typescript-estree/src/convert-comments.ts#L147-L154

These cases need to not get caught in here.
